bokehjs/src/lib/models/ranges/factor_range.ts

```diff
@@ -147,14 +199,14 @@ export class FactorRange extends Range {
       end:                 [ Float, p.unset, {readonly: true} ],
     }))
 
-    this.internal<FactorRange.Props>(({Float, Str, List, Tuple, Nullable}) => ({
-      levels: [ Float ], // how many levels of
+    this.internal<FactorRange.Props>(({Int, Str, List, Tuple, Nullable}) => ({
+      levels: [ Int ], // how many levels of nesting
       mids:   [ Nullable(List(Tuple(Str, Str))), null ], // mid level factors (if 3 total levels)
       tops:   [ Nullable(List(Str)), null ], // top level factors (whether 2 or 3 total levels)
```

Comment:
@mattpap do these actually need to be Bokeh properties?

Reply:
I guess things like categorical axis and ticker access them, but it's still not clear to me that they need to be actual properties for that.

Reply:
That's how these were implemented originally, so that's how the have been kept since. Looking at their usage, there is no reason for them to continue being properties. The primary reason would be any change event listeners, but there are none currently.